### DISCUSSION TITLE
How can I create a PR for all the branches in a stack?

### DISCUSSION THREAD

**Discussion by michaelschufi** [git-town/git-town#3839] [Q&A] [answered]
Hi

While looking for an alternative to Graphite, I found git town. The stacking works so far, I think.

I currently have 3 branches in a stack that I would like to open PRs for. When executing `git town propose` only the PR for the top branch to the 2nd top branch is being created.

E.g. for 
```
main
 \
  1-fix-typos
   \
    2-rename-foo
     \
      3-rename-bar
```
`git town propose` while on `3-rename-bar` opens the PR for `3-rename-bar` -> `2-rename-foo`.

Is there a way to automatically create the PRs for the whole stack if they don't exist yet?

**Comment by kevgo** [comment, marked as answer]
That's a great idea! Not implemented yet. A possible syntax for this would be

```
git town propose --stack
```

This would match the already existing `git sync --stack` (which syncs all branches in the stack) and `git town compress --stack` (which compresses all branches in the stack).

I have created https://github.com/git-town/git-town/issues/3840 to track this idea in the backlog. Thanks for the great suggestion! 🙏
